### ISSUE TITLE
[REQ] Use Autoprefixer on CSS optimization stream

### ISSUE THREAD

**Issue by stramel** [Polymer/polymer-cli#725]
Using Autoprefixer on the CSS optimization stream.

> It is recommended by Google and used in Twitter and Taobao.

This will allow a user to use a `browserslist` file to set their support and clean up any unnecessary prefixes or add any missed prefixes. This is great for shipping less code and the only piece missing from my entire gulpfile. 

I *believe* without a `browserslist` it leaves the code untouched. You *may* be able to provide defaults for what Polymer supports. 

The use of `browserslist` would also be awesome if you guys decide to add something like `babel-preset-env` which also uses the list to only transpile what the browser doesn't support already.

**Comment by jsilvermist** [comment]
+1, but instead of leaving the code untouched I thought it would default to something like: >1% & last 2 versions?

**Comment by stramel** [comment]
@jsilvermist it may be implementation specific, I didn't see anything at first glace in the core autoprefixer code. Either way, it should still work with `browserslist` :)

**Comment by abdonrd** [comment]
Would be nice if the CLI "compiles" the CSS as the Javascript.

So I guess we also remove the prefixes in elements like the [PolymerElements/iron-flex-layout.html#L66-L67](https://github.com/PolymerElements/iron-flex-layout/blob/2.0-preview/iron-flex-layout.html#L66-L67).

**Comment by davidmaxwaterman** [comment]
Personally, I'd go for css-next (it includes autoprefixer), but I suppose that might interfere with some of polymer's css syntax, though perhaps there is scope for Polymer to be smaller?

http://cssnext.io/features/#automatic-vendor-prefixes

**Comment by stramel** [comment]
@abdonrd Exactly, autoprefixer adds missing prefixes and removes unnecessary prefixes. So the lines you highlighted in `iron-flex-layout` would be removed if they weren't necessary based on your `browserslist` file or left untouched if no `browserslist` file exists.

@davidmaxwaterman use of cssnext is cool however it is essentially polyfilling/shimming these functionalities. I could possibly see them implementing postCSS and allowing the user to configure the plugins to use, maybe?

**Comment by davidmaxwaterman** [comment]
@stramel I don't think it's a polyfill, though it seems it can be used that way. Shimming, yes, but it's all done via preprocessing, just like the vendor prefixing. Well, that's how I've always used it in the past, though I've only really used it for specific features (like variables, and prefixing) and haven't fully explored everything.

I agree, though, that flexibility is the best solution ie I am enthusiastic about supporting postcss and letting the user decide what plugins to use.

**Comment by mgibas** [comment]
You can run postcss as a part of build step - take a look at my comment in #243

**Comment by mgibas** [comment]
Yeah, in my case I dont care about auto prefixing till deployment so I can run that on CI
